workspaces/optic-engine/tests/interaction_diff.rs

```diff
@@ -240,15 +255,29 @@ async fn can_yield_unmatched_shape() {
   )
   .expect("example http interaction should deserialize");
 
-  let mut results = diff_interaction(
+  let results = diff_interaction(
     &spec_projection,
     compliant_interaction,
     &DiffInteractionConfig::default(),
   );
   assert_debug_snapshot!(results);
   assert_eq!(results.len(), 0);
+}
 
-  let incompliant_interaction = HttpInteraction::from_json_str(
+#[test]
+fn can_handle_no_request_and_response_body() {
```

Comment:
New test, this includes events that are created when we learn an interaction with no request or response body (i.e. there are no shapes)